utils/vpinns/quad_rule.py

```diff
@@ -147,17 +148,25 @@ def get_test_function_properties(n_test, coord_quadrature, approach="2"):
     """
     
     dict_legendre = {"1" : modified_legendre,
-                     "2" : modified_legendre_2}
+                     "2" : modified_legendre_2,
+                     "3" : legendre_basis_shape_functions,
+                     "4" : lagrange_basis_shape_functions}
```

Comment:
Just a general remark:

Since you have the dictionary here anyway, I would go for more descriptive keys for the different options, i.e., something like 

```python
dict_legendre = {
                      "modified_legendre_scipy" : modified_legendre,
                      "modified_legendre_custom" : modified_legendre_2,
                      "legendre_basis_custom" : legendre_basis_shape_functions,
                      "lagrange_basis_custom" : lagrange_basis_shape_functions
                      }
```

That way one would directly know without looking into the source code where the difference between the different options is.